### PR TITLE
Panasonic DMC-LX15: Added basic camera support

### PR DESCRIPTION
**Pull request by schenlap**
Samples are from
http://www.photographyblog.com/reviews/panasonic_lumix_dmc_lx15_review/sample_images/

### CODE REVIEW

**Comment by schenlap** [comment]
black levels look strange, but i got this results....

**Comment by LebedevRI** [comment]
> black levels look strange, but i got this results....

That is actually more plausible than single black level for all ISO's

File: src/external/rawspeed/data/cameras.xml [hunk 1]
```diff
@@ -5734,6 +5734,40 @@
 		<Crop x="0" y="0" width="-74" height="0"/>
 		<Sensor black="15" white="3971"/>
 	</Camera>
+	<Camera make="Panasonic" model="DMC-LX15">
+		<ID make="Panasonic" model="DMC-LX15">Panasonic DMC-LX15</ID>
```
Comment (LebedevRI) [inline src/external/rawspeed/data/cameras.xml]: I believe http://www.imaging-resource.com/PRODS/panasonic-lx10/panasonic-lx10A.HTM says that LX9, LX10 and LX15 are the same camera. And that page has DMC-LX10 sample.

File: src/external/rawspeed/data/cameras.xml [hunk 2]
```diff
@@ -5734,6 +5734,40 @@
...
+		<CFA width="2" height="2">
+			<Color x="0" y="0">GREEN</Color>
+			<Color x="1" y="0">BLUE</Color>
+			<Color x="0" y="1">RED</Color>
+			<Color x="1" y="1">GREEN</Color>
+		</CFA>
+		<Crop x="0" y="0" width="0" height="0"/>
+		<Sensor black="142" white="4095" iso_list="80 100 125 400"/>
```
Comment (LebedevRI) [inline src/external/rawspeed/data/cameras.xml]: Something is wrong here, i get different results, especially for ISO80, especially for ISO80 whitelevel.

```
../tools/dngmeta.rb lx15/*.DNG
invsensors {[142, 2095]=>[80], [142, 3400]=>[100], [142, 4095]=>[125], [143, 4095]=>[160, 200, 250, 320, 400], [144, 4095]=>[640, 800], [145, 4095]=>[1250, 1600], [149, 4095]=>[3200], [154, 4095]=>[6400], [166, 4095]=>[12800], [167, 4095]=>[25600]}

sensors [[80, [142, 2095]], [100, [142, 3400]], [125, [142, 4095]], [160, [143, 4095]], [200, [143, 4095]], [250, [143, 4095]], [320, [143, 4095]], [400, [143, 4095]], [640, [144, 4095]], [800, [144, 4095]], [1250, [145, 4095]], [1600, [145, 4095]], [3200, [149, 4095]], [6400, [154, 4095]], [12800, [166, 4095]], [25600, [167, 4095]]]
mostfrequent [[143, 4095], [160, 200, 250, 320, 400]]
invsensors {[142, 2095]=>[80], [142, 3400]=>[100], [142, 4095]=>[125], [144, 4095]=>[640, 800], [145, 4095]=>[1250, 1600], [149, 4095]=>[3200], [154, 4095]=>[6400], [166, 4095]=>[12800], [167, 4095]=>[25600]}

        <Camera make="Panasonic" model="DMC-LX15">
                <ID make="" model="">Panasonic DMC-LX9</ID>
                <Sensor black="143" white="4095"/>
                <Sensor black="142" white="2095" iso_list="80"/>
                <Sensor black="142" white="3400" iso_list="100"/>
                <Sensor black="142" white="4095" iso_list="125"/>
                <Sensor black="144" white="4095" iso_list="640 800"/>
                <Sensor black="145" white="4095" iso_list="1250 1600"/>
                <Sensor black="149" white="4095" iso_list="3200"/>
                <Sensor black="154" white="4095" iso_list="6400"/>
                <Sensor black="166" white="4095" iso_list="12800"/>
                <Sensor black="167" white="4095" iso_list="25600"/>
        </Camera>
```

Comment (schenlap) [inline src/external/rawspeed/data/cameras.xml]: Ok, whitelevels are my error, but:

with ISO1600 samples i get different results depending on the tool i use:
dngmeta.sh panasonic_lumix_dmc_lx15_68.dng: Sensor black="**146**" white="4095"/

dngmeta.rb panasonic_lumix_dmc_lx15_68.dng: 
```
sensors [[1600, [145, 4095]]]
mostfrequent [[145, 4095], [1600]]
invsensors {}

        <Camera make="Panasonic" model="DMC-LX15">
                <ID make="" model="">Panasonic DMC-LX9</ID>
                <Sensor black="145" white="4095"/>
        </Camera>
```

Comment (LebedevRI) [inline src/external/rawspeed/data/cameras.xml]: Yes, `dngmeta.rb` is non-deterministic in that regard, https://github.com/darktable-org/darktable/blob/62416056d956a05f891846dc78fc11033d3c699b/tools/dngmeta.rb#L89-L92
If you change `BLACKDIFF_MAX` to `0`, the script then should fail with an error message.
I'm not 100% sure what `dngmeta.rb` should do else if there is more than one black level for a given ISO.

Comment (schenlap) [inline src/external/rawspeed/data/cameras.xml]: it does not fail with BLACKDIFF_MAX to 0.

Comment (LebedevRI) [inline src/external/rawspeed/data/cameras.xml]: > For one file it should return the same value as dngmeta.sh, or?

Yes, i expect that when called for a single DNG, it should provide the same output as `dngmeta.sh`

BTW: i plan on eventually dropping `dngmeta.sh`, once `dngmeta.rb` fully supersedes it. So if `dngmeta.rb` has any bugs, better raise them in redmine.

Comment (LebedevRI) [inline src/external/rawspeed/data/cameras.xml]: > it does not fail with BLACKDIFF_MAX to 0.

Hm, i did not check before writing. It does not fail, but it does warn and produce intentionally-wrong results:
```
ISO 125 has multiple variants with too different black levels: [[141, 4095], [142, 4095]]
ISO 160 has multiple variants with too different black levels: [[142, 4095], [143, 4095]]
ISO 250 has multiple variants with too different black levels: [[142, 4095], [143, 4095]]
ISO 400 has multiple variants with too different black levels: [[142, 4095], [143, 4095]]
ISO 800 has multiple variants with too different black levels: [[142, 4095], [144, 4095]]
ISO 1600 has multiple variants with too different black levels: [[141, 4095], [145, 4095]]
invsensors {[142, 2095]=>[80], [142, 3400]=>[100], [-1, -1]=>[125, 160, 250, 400, 800, 1600], [143, 4095]=>[200, 320], [144, 4095]=>[640], [145, 4095]=>[1250], [149, 4095]=>[3200], [154, 4095]=>[6400], [166, 4095]=>[12800], [167, 4095]=>[25600]}

sensors [[80, [142, 2095]], [100, [142, 3400]], [125, [-1, -1]], [160, [-1, -1]], [200, [143, 4095]], [250, [-1, -1]], [320, [143, 4095]], [400, [-1, -1]], [640, [144, 4095]], [800, [-1, -1]], [1250, [145, 4095]], [1600, [-1, -1]], [3200, [149, 4095]], [6400, [154, 4095]], [12800, [166, 4095]], [25600, [167, 4095]]]
mostfrequent [[-1, -1], [125, 160, 250, 400, 800, 1600]]
invsensors {[142, 2095]=>[80], [142, 3400]=>[100], [143, 4095]=>[200, 320], [144, 4095]=>[640], [145, 4095]=>[1250], [149, 4095]=>[3200], [154, 4095]=>[6400], [166, 4095]=>[12800], [167, 4095]=>[25600]}

        <Camera make="Panasonic" model="DMC-LX15">
                <ID make="" model="">Panasonic DMC-LX9</ID>
                <Sensor black="-1" white="-1"/>
                <Sensor black="142" white="2095" iso_list="80"/>
                <Sensor black="142" white="3400" iso_list="100"/>
                <Sensor black="143" white="4095" iso_list="200 320"/>
                <Sensor black="144" white="4095" iso_list="640"/>
                <Sensor black="145" white="4095" iso_list="1250"/>
                <Sensor black="149" white="4095" iso_list="3200"/>
                <Sensor black="154" white="4095" iso_list="6400"/>
                <Sensor black="166" white="4095" iso_list="12800"/>
                <Sensor black="167" white="4095" iso_list="25600"/>
        </Camera>
```
Note the `black="-1" white="-1"`

**Comment by LebedevRI** [comment]
Btw, since there is no crop, there is obiviously no black areas, so it means there must be some exif tag that rawspeed decoder should read and use as proper blacklevel.

**Comment by LebedevRI** [comment]
@schenlap thank you.

**Pull request merged**